src/qililab/qprogram/quantum_machines_compiler.py

```diff
@@ -137,6 +137,7 @@ def __init__(self) -> None:
             ResetPhase: self._handle_reset_phase,
             Wait: self._handle_wait,
             Sync: self._handle_sync,
+            Block: self._handle_block,
```

Comment:
We should add a test for this as well: Wrap a fixture with the `with qp.block:` and assert the QUA remains unchanged.

Reply:
yes, I was just adding tests but have no finished yet, also the unittest for QM has not been done yet

Reply:
Tests done @fedonman